pkg/auth/handler/get_role.go

```diff
@@ -0,0 +1,79 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/skygeario/skygear-server/pkg/auth"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authinfo"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authz"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authz/policy"
+	"github.com/skygeario/skygear-server/pkg/core/handler"
+	"github.com/skygeario/skygear-server/pkg/core/inject"
+	"github.com/skygeario/skygear-server/pkg/core/server"
+	"github.com/skygeario/skygear-server/pkg/server/skyerr"
+)
+
+func AttachGetRoleHandler(
+	server *server.Server,
+	authDependency auth.DependencyMap,
+) *server.Server {
+	server.Handle("/role/get", &GetRoleHandlerFactory{
+		authDependency,
+	}).Methods("POST")
+	return server
+}
+
+type GetRoleHandlerFactory struct {
+	Dependency auth.DependencyMap
+}
+
+func (f GetRoleHandlerFactory) NewHandler(request *http.Request) http.Handler {
+	h := &GetRoleHandler{}
+	inject.DefaultInject(h, f.Dependency, request)
+	return handler.APIHandlerToHandler(h)
+}
+
+func (f GetRoleHandlerFactory) ProvideAuthzPolicy() authz.Policy {
+	return policy.AllOf(
+		// FIXEM: this endpoint should hanlde request with master key or with access key
+		// Users can only get his own roles except that administrators can query roles
+		// of other users.
+		// This is temporary implementation to support admin user role only.
+		authz.PolicyFunc(policy.RequireMasterKey),
+		authz.PolicyFunc(policy.RequireAuthenticated),
+		authz.PolicyFunc(policy.DenyDisabledUser),
+	)
+}
+
+type GetRoleRequestPayload struct {
+	UserIDs []string `json:"users"`
+}
+
+func (p GetRoleRequestPayload) Validate() error {
+	return nil
+}
+
+// GetRoleHandler handles signup request
```

Comment:
// GetRoleHandler handles signup request
->
// GetRoleHandler allow admin user to query user roles.

Reply:
Please add an example input and outpur as in https://github.com/SkygearIO/skygear-server/blob/next/pkg/auth/handler/role_assign.go#L62